wallets/react-wallet-chat/src/components/SessionProposalChainCard.tsx

```diff
@@ -31,16 +31,15 @@ interface IProps {
 export default function SessionProposalChainCard({ requiredNamespace }: IProps) {
   return (
     <Fragment>
-      {requiredNamespace.chains.map(chainId => {
+      {requiredNamespace.chains?.map(chainId => {
         const extensionMethods: ProposalTypes.RequiredNamespace['methods'] = []
         const extensionEvents: ProposalTypes.RequiredNamespace['events'] = []
 
-        requiredNamespace.extension?.map(({ chains, methods, events }) => {
-          if (chains.includes(chainId)) {
-            extensionMethods.push(...methods)
-            extensionEvents.push(...events)
-          }
-        })
+        const { chains, methods, events } = requiredNamespace
```

Comment:
Can also be removed, `allMethods` and `extensionMethods` (same with events) is no longer needed, can use `requiredNamespace.methods` directly below